### PR TITLE
DP-6747 / DP-6746

### PR DESCRIPTION
**Pull request by mtpaasto**
Still to be tested

### CODE REVIEW

File: modules/mesh_com/entrypoint.sh [hunk 1]
```diff
@@ -1,62 +1,72 @@
...
+elif [ "$DRONE_TYPE" == "cm-fog" ]; then
+    DEFAULT_MESH_IP="192.168.248.$((RANDOM % 243 + 11))"
+    gateway_ip="192.168.248.1" # FIXME: hardcoded for now. later detect automatically.
+    
+else
+    echo "drone type not implemented: $DRONE_TYPE"
+    exit 1
+fi
+
+/opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
```
Comment (maseabunikie) [inline modules/mesh_com/entrypoint.sh:61]: mesh-executor should be enabled here. See row: https://github.com/tiiuae/mesh_com/pull/309/files#diff-31f2e5de9f3f4a563b757fc677f251e921209d421672608b52d862e6f73f6cf5R96

File: modules/mesh_com/entrypoint.sh [hunk 2]
```diff
@@ -1,62 +1,76 @@
...
-            route add default gw $gateway_ip bat0
-        fi
-        sleep 86400
-    elif [ "$DRONE_TYPE" == "singlemesh" ]; then
-        # 192.168.248.11-192.168.248.253
-        DEFAULT_MESH_IP="192.168.248.$[ $RANDOM % 243 + 11 ]"
-        /opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
-        echo "mesh setup done"
-            # mesh class is gs
+if [ "$DRONE_TYPE" == "cm-recon" ]; then
```
Comment (maseabunikie) [inline modules/mesh_com/entrypoint.sh:35]: Please, fix indentation of these if clauses.

**Review by maseabunikie — APPROVED** [review]
Well done! Let's take this into tests.

**Comment by DamienDeMayaTII** [comment]
@mtpaasto @maseabunikie, I am not sure what you mean by let's take this into tests. If the code is not tested it cannot be in MS Develop. And if it is tested then merge and let's not keep it open. Thanks

**Comment by maseabunikie** [comment]
@DamienDeMayaTII This PR is merging changes that need to be applied to feat-multiach branch, not develop. We need to merge/rebase changes before we can test. If you can see the timeline, the approval from me came last evening and this morning we did not have time yet to do the merge into feat-multiarch.

**Comment by maseabunikie** [comment]
No problem. :smile:

**Pull request merged**